rtsx.h

```diff
@@ -169,22 +175,34 @@ static inline struct rtsx_dev *host_to_rtsx(struct Scsi_Host *host) {
 
 static inline void get_current_time(u8 *timeval_buf, int buf_len)
 {
-	struct timeval tv;
+	struct TIMESTRUCT tv;
 
 	if (!timeval_buf || (buf_len < 8)) {
 		return;
 	}
 
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 0, 0)
 	do_gettimeofday(&tv);
+#else
+	ktime_get_real_ts64(&tv);
+#endif
 
 	timeval_buf[0] = (u8)(tv.tv_sec >> 24);
 	timeval_buf[1] = (u8)(tv.tv_sec >> 16);
 	timeval_buf[2] = (u8)(tv.tv_sec >> 8);
 	timeval_buf[3] = (u8)(tv.tv_sec);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 0, 0)
 	timeval_buf[4] = (u8)(tv.tv_usec >> 24);
 	timeval_buf[5] = (u8)(tv.tv_usec >> 16);
 	timeval_buf[6] = (u8)(tv.tv_usec >> 8);
 	timeval_buf[7] = (u8)(tv.tv_usec);
+#else
+	timeval_buf[4] = (u8)(tv.tv_nsec >> 24);
```

Comment:
They're not:

```
long int tv_usec
This is the rest of the elapsed time (a fraction of a second), represented as the number of microseconds. It is always less than one million.
```
(source: https://www.gnu.org/software/libc/manual/html_node/Elapsed-Time.html)

```
long tv_nsec	nanoseconds (valid values are [0, 999999999])
```
(source: https://en.cppreference.com/w/c/chrono/timespec)

But then again this structure members are used only to construct an opaque buffer of bytes rather than a meaningful structure, which in turn is used for tracing which looks like a debug feature at a first glance. So I'm not sure if a proper multiplication is desired to begin with?